ui/src/app/utils/runtime-utils.tsx

```diff
@@ -133,11 +344,33 @@ export const useCustomRuntime = (currentWorkspaceNamespace):
       // TODO: It is likely more correct here to use the LeoRuntimeInitializer wait for the runtime
       // to reach a terminal status before attempting deletion.
       try {
-        if (runtime && runtime.status !== RuntimeStatus.Deleted) {
-          await runtimeApi().deleteRuntime(currentWorkspaceNamespace, {
-            signal: aborter.signal
-          });
+        if (runtime) {
+          const runtimeDiffTypes = getRuntimeDiffs(runtime, requestedRuntime).map(diff => diff.differenceType);
+
+          if (runtimeDiffTypes.includes(RuntimeDiffState.NEEDS_DELETE)) {
+            if (runtime.status !== RuntimeStatus.Deleted) {
+              await runtimeApi().deleteRuntime(currentWorkspaceNamespace, {
+                signal: aborter.signal
+              });
+            }
+          } else if (runtimeDiffTypes.includes(RuntimeDiffState.CAN_UPDATE)) {
+            if (runtime.status === RuntimeStatus.Running || runtime.status === RuntimeStatus.Stopped) {
+              await runtimeApi().updateRuntime(currentWorkspaceNamespace, {runtime: requestedRuntime});
+              // Calling updateRuntime will not immediately set the Runtime status to not Running so the
+              // default initializer will resolve on its first call. The polling below first checks for the
+              // non Running status before initializing the default one that checks for Running status
+              await LeoRuntimeInitializer.initialize({
```

Comment:
Does this interact strangely with the call on L374?

Reply:
the initializer on  L374 should only start after this one has completed. (intentionally)